src/modules/dynamic-ui-wrapper/dynamic-ui-wrapper.ts

```diff
@@ -7,24 +7,23 @@ import {IIdentity} from '@essential-projects/iam_contracts';
 import {DataModels} from '@process-engine/management_api_contracts';
 
 import {
-  IBooleanFormField,
   IDynamicUiService,
-  IEnumFormField,
   ISolutionEntry,
-  IStringFormField,
 } from '../../contracts';
 
+enum ButtonClickActions {
+  cancel = 'cancel',
+  process = 'proceed',
+  decline = 'decline',
+}
+
 @inject('DynamicUiService', Router, Element)
 export class DynamicUiWrapper {
 
-  public cancelButtonText: string = 'Cancel';
-  public confirmButtonText: string = 'Continue';
-  public declineButtonText: string = 'Decline';
-  public onButtonClick: (action: 'cancel' | 'proceed' | 'decline') => void;
-  @bindable({changeHandler: 'userTaskChanged'}) public currentUserTask: DataModels.UserTasks.UserTask;
-  @bindable({changeHandler: 'manualTaskChanged'}) public currentManualTask: DataModels.ManualTasks.ManualTask;
-  @bindable() public isConfirmUserTask: boolean = false;
-  @bindable() public isFormUserTask: boolean = false;
+  public onButtonClick: (action: ButtonClickActions) => void;
```

Comment:
So the pattern here is, that the DynamicUiWrapper contains a TaskDynamicUi, which will set this function at some point with: 
```ts
    this.dynamicUiWrapper.onButtonClick = (action: string): void => {
      this._finishTask(action);
    };
```
Correct me if I'm wrong, that was my understanding.

I find that callback-pattern confusing and it has the downside that onButtonClick is not set at all times.
I would recommend using Events instead.

If this must stay, I would add the `undefined` type.

```ts
   public onButtonClick: ((action: ButtonClickActions) => void) | undefined;
```

Reply:
I don't think adding `| undefined` would be helpful.
Any object can at any time be undefined, regardless of its type. But we don't include undefined in the type signatures. Why would we?
If we did, we'd have to write all typedefs like so:

```ts
  @bindable() public currentUserTask: DataModels.UserTasks.UserTask | undefined;
  @bindable() public currentManualTask: DataModels.ManualTasks.ManualTask | undefined;
``` 

Because these objects could also be undefined at some point.

I think just leaving the typedef as it is and performing a check for "undefined", prior to using it, suffices.
Such a check is performed here, for example:
https://github.com/process-engine/bpmn-studio/blob/6a275aaed22217e5c2df4f52d9a5f3678cf2e6e3/src/modules/dynamic-ui-wrapper/dynamic-ui-wrapper.ts#L121

But I understand your confusion.
Perhaps an explanatory comment on top of this callback might clear this up a bit.

Dunno, maybe something like 
```suggestion
  /**
   * This callback is set by the TaskDynamicUi.
   */
  public onButtonClick: (action: ButtonClickActions) => void;
```

Reply:
I realize that we would have to use `| undefined` in many cases and I honestly think that would be good.

If possible it's a good thing to initialize variables. A variable not being set immediatly should be the special case. 
TypeScript does not really help you to avoid all the `cannot read property _ of undefined` errors and all the `|| thing === undefined || thing === null`-checks unless you differentiate between types which are always set and types which are maybe set.
If something is `undefined` at runtime it should have a reason for that. In case of the binding examples you gave I think it does make sense. In the constructor you cannot access those properties, so the type `DataModels.UserTasks.UserTask` is not reliable.
You could also make `type Maybe<a> = a | undefined;`. Looks a bit like haskell then.

I don't want to add that type everywhere, I want to have the variables initialized with the right value if possible(and eliminate null).

I realize now that my opinion may be opinionated ;)
Just wanted to clarify my motivation.